packages/core/src/utils/tokens/get-token-from-ans.ts

```diff
@@ -0,0 +1,54 @@
+import request from 'graphql-request'
+import { gql } from '../../codegen/gql'
+import { AssetType } from '../../codegen/gql/graphql'
+import { CW20Token } from './cw20/cw20-token'
+import { NativeToken } from './native/native-token'
+
+export async function getTokenFromAns(
+  ansName: string,
+  chain: string,
+  apiURL: string,
+) {
+  const result = await request(
+    apiURL,
+    gql(
+      `
+  query AssetsQuery($chain: ID!, $filter: IdsFilter!) {
+    ans(chain: $chain) {
+      assets(filter: $filter) {
```

Comment:
Let's make a task to deprecate the `assets` api to be replaced with `tokens`.

Reply:
We can improve it at the same time to return different types based on cw20 or native, getting rid of the overridden "address"

Reply:
https://linear.app/abstract-sdk/issue/APP-264/refactor-the-asset-field-to-token-on-subgraph